Rules.py

```diff
@@ -66,5 +66,62 @@ def global_rules(world):
+
+    # obviously, this and simliar 'tests' might not be necessary at all
+    set_rule(world.get_location('Adult Wallet from bank'), lambda state: True)
+    set_rule(world.get_location('Bank HP'), lambda state: True)
+
+    # I don't know that this item matters at all or if we should even have a spot for it
+    # but we can if we want, you get it the same way you get the code in the first cycle, you just have to be human to
+    # get the actual notebook from them I think
+    set_rule(world.get_location('Bomber Notebook'), lambda state: state.has('Magic Meter') and state.form('Deku') and state.form('Human'))
+
+    # since getting the blast mask just involves slashing sakon, it might be possible to get it with other forms? like
+    # maybe zora? probly not goron though
+    # also you might need to be link to talk to the lady after, so maybe it is only human
+    set_rule(world.get_location('Blast Mask'), lambda state: state.form('Human'))
+
+    set_rule(world.get_location('North Clock Town Tree HP'), lambda state: True)
+    set_rule(world.get_location('Tingle Clock Town Map'), lambda state: True)
+    set_rule(world.get_location('Tingle Woodfall Map'), lambda state: True)
+    set_rule(world.get_location('Keaton HP'), lambda state: state.has('Keaton Mask'))
+
+    # I need to figure out all the actual requirements for this, including tricks, so this one is tentative
+    # also the trick names are guesses for sure
+    set_rule(world.get_location('East Clock Town 100 Rupee Chest'), lambda state: state.form('Human') or state.can('Goron Boost') or state.can('Gainer'))
+
+    set_rule(world.get_location('Ocarina of Time'), lambda state: (state.form('Deku') and state.has('Magic Meter')) or (state.has('Bow') and state.form('Human')))
+    set_rule(world.get_location('All Night mask'), lambda state: state.form('Human') and state.has('Giants Wallet'))
+    set_rule(world.get_location('Bigger Bomb Bag'), lambda state: state.has('Giants Wallet'))
+
+    # you might not need to be human to get this, but I'd bet the shop owner won't sell to other forms, at least deku
+    set_rule(world.get_location('Bomb Bag'), lambda state: state.form('Human'))
+
+    set_rule(world.get_location('Sword School HP'), lambda state: state.form('Human'))
+
+    # i swear this should be an optional trick, it's so hard without the bunny hood lol
+    set_rule(world.get_location('Postman Game HP'), lambda state: state.can('Time ten') or state.has('Bunny Hood'))
+
+    set_rule(world.get_location('Deku Scrub Playground HP'), lambda state: state.form('Deku'))
+    set_rule(world.get_location('Great Fairy Mask'), lambda state: True)
+    set_rule(world.get_location('Clock Town Maze Minigame HP'), lambda state: state.form('Goron'))
+    set_rule(world.get_location('Bomber Hideout 100 Rupee Chest'), lambda state: state.has('Bomber Code') and (state.has('Bomb Bag') or state.has('Blast Mask')))
```

Comment:
You technically don't need the Bomber Code, only access to the Bomber Tunnel/Astral Observatory. (and you can get in from Termina Field under certain conditions)